From 66759f19ded731824c46d9b01cbef52c81e41209 Mon Sep 17 00:00:00 2001
From: Marek Lindner <lindner_marek@yahoo.de>
Date: Thu, 2 Apr 2009 20:52:36 +0200
Subject: [PATCH] Add forgotten rcu for if_list

---
 hard-interface.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/hard-interface.c b/hard-interface.c
index e5dbefc..c182c79 100644
--- a/hard-interface.c
+++ b/hard-interface.c
@@ -48,7 +48,7 @@ int hardif_min_mtu(void)
 	 * (have MTU > 1500 + BAT_HEADER_LEN) */
 	int min_mtu = 1500;	
 	rcu_read_lock();
-	list_for_each_entry(batman_if, &if_list, list) {
+	list_for_each_entry_rcu(batman_if, &if_list, list) {
 		if (batman_if->if_active == IF_ACTIVE) 
 			min_mtu = MIN(batman_if->net_dev->mtu - BAT_HEADER_LEN, min_mtu);
 	}
-- 
1.6.2.1

