From b170b1d8c8601139dd2efe1f05ee68c919aa9d84 Mon Sep 17 00:00:00 2001
From: Simon Wunderlich <siwu@hrz.tu-chemnitz.de>
Date: Thu, 2 Apr 2009 12:34:51 +0200
Subject: [PATCH] restore irqs when putting a character because __put_user might sleep (thanks sven)

---
 log.c |    5 +++--
 1 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/log.c b/log.c
index 400dbce..3892e31 100644
--- a/log.c
+++ b/log.c
@@ -148,14 +148,15 @@ ssize_t log_read(struct file *file, char __user *buf, size_t count, loff_t *ppos
 
 		log_start++;
 
-		spin_unlock(&logbuf_lock);
+		spin_unlock_irqrestore(&logbuf_lock, flags);
 
 		error = __put_user(c,buf);
 
+		spin_lock_irqsave(&logbuf_lock, flags);
+
 		buf++;
 		i++;
 
-		spin_lock(&logbuf_lock);
 	}
 
 	spin_unlock_irqrestore(&logbuf_lock, flags);
-- 
1.6.2.1

