Description: Check icmp for version before sending
Origin: backport, http://open-mesh.net/changeset/1314
Author: Marek Lindner <lindner_marek@yahoo.de>

---
diff --git a/device.c b/device.c
index dca1921b50717565959e87d75aad147905b35d8c..99556b06a9d664b9078a5e1886586bfa77d63b0b 100644
--- a/device.c
+++ b/device.c
@@ -233,6 +233,15 @@ ssize_t bat_device_write(struct file *file, const char __user *buff, size_t len,
 		return -EFAULT;
 	}
 
+	icmp_packet.uid = device_client->index;
+
+	if (icmp_packet.version != COMPAT_VERSION) {
+		icmp_packet.msg_type = PARAMETER_PROBLEM;
+		icmp_packet.ttl = COMPAT_VERSION;
+		bat_device_add_packet(device_client, &icmp_packet);
+		goto out;
+	}
+
 	if ((icmp_packet.packet_type == BAT_ICMP) && (icmp_packet.msg_type == ECHO_REQUEST)) {
 
 		spin_lock(&orig_hash_lock);
@@ -241,14 +250,12 @@ ssize_t bat_device_write(struct file *file, const char __user *buff, size_t len,
 		if ((orig_node != NULL) && (orig_node->batman_if != NULL) && (orig_node->router != NULL)) {
 
 			memcpy(icmp_packet.orig, orig_node->batman_if->net_dev->dev_addr, ETH_ALEN);
-			icmp_packet.uid = device_client->index;
 
 			send_raw_packet((unsigned char *)&icmp_packet, sizeof(struct icmp_packet), orig_node->batman_if->net_dev->dev_addr, orig_node->router->addr, orig_node->batman_if);
 
 		} else {
 
 			icmp_packet.msg_type = DESTINATION_UNREACHABLE;
-			icmp_packet.uid = device_client->index;
 			bat_device_add_packet(device_client, &icmp_packet);
 
 		}
@@ -262,6 +269,7 @@ ssize_t bat_device_write(struct file *file, const char __user *buff, size_t len,
 
 	}
 
+out:
 	return len;
 }
 
diff --git a/packet.h b/packet.h
index 3c6c23bbf9c665687aa082d9aea2e300031f6ef1..b4b2528a5073756588d004d21f75999d56e31a17 100644
--- a/packet.h
+++ b/packet.h
@@ -29,11 +29,12 @@
 
 
 
+/* ICMP message types */
 #define ECHO_REPLY 0
 #define DESTINATION_UNREACHABLE 3
 #define ECHO_REQUEST 8
 #define TTL_EXCEEDED 11
-
+#define PARAMETER_PROBLEM 12
 
 
 struct batman_packet
