Description: Fix half copied icmp paket to userspace
Origin: backport, http://open-mesh.net/changeset/1329
Author: Marek Lindner <lindner_marek@yahoo.de>

---
diff --git a/device.c b/device.c
index 028cf7df5c5749b6ced4add7375b44077e517215..dca1921b50717565959e87d75aad147905b35d8c 100644
--- a/device.c
+++ b/device.c
@@ -229,7 +229,7 @@ ssize_t bat_device_write(struct file *file, const char __user *buff, size_t len,
 	if (!access_ok(VERIFY_READ, buff, sizeof(struct icmp_packet)))
 		return -EFAULT;
 
-	if (__copy_from_user(&icmp_packet, buff, sizeof(icmp_packet))) {
+	if (__copy_from_user(&icmp_packet, buff, sizeof(struct icmp_packet))) {
 		return -EFAULT;
 	}
 
@@ -287,7 +287,7 @@ void bat_device_add_packet(struct device_client *device_client, struct icmp_pack
 		return;
 
 	INIT_LIST_HEAD(&device_packet->list);
-	memcpy(&device_packet->icmp_packet, icmp_packet, sizeof(icmp_packet));
+	memcpy(&device_packet->icmp_packet, icmp_packet, sizeof(struct icmp_packet));
 
 	spin_lock(&device_client->lock);
 
