Description: Add forgotten rcu for if_list
Origin: backport, http://open-mesh.net/changeset/1216
Author: Marek Lindner <lindner_marek@yahoo.de>

---
diff --git a/hard-interface.c b/hard-interface.c
index e5dbefc3be03f9919675490598176da5bf1e9e39..c182c7998987153b52d4c5ff6c6d2beeeda1159b 100644
--- a/hard-interface.c
+++ b/hard-interface.c
@@ -48,7 +48,7 @@ int hardif_min_mtu(void)
 	 * (have MTU > 1500 + BAT_HEADER_LEN) */
 	int min_mtu = 1500;	
 	rcu_read_lock();
-	list_for_each_entry(batman_if, &if_list, list) {
+	list_for_each_entry_rcu(batman_if, &if_list, list) {
 		if (batman_if->if_active == IF_ACTIVE) 
 			min_mtu = MIN(batman_if->net_dev->mtu - BAT_HEADER_LEN, min_mtu);
 	}
